docs/testplan/VxLAN_BGP_EVPN_test_plan.md

```diff
@@ -0,0 +1,225 @@
+* Verify that different route types are learned and shown in database.
+* Send traffic from local host to remote hosts configured behind these VTEPs.
+* Match the tunnel counters with traffic statistics.
+* Enable egress tracking on vlan and see packets mapped between VNI and Vlan's.
+* Traffic should flow without any loss at line rate.
+* Do warm, fast, soft and cold reboots while traffic is running.
+
+#### Test results
+
+
+### Test case # 4 – Validate the functionality of L2VNI and the maximum L2VNIs supported per tunnel - 4K.
+#### Test objective
+Verify that switch supports total 4K L2 VNI's per tunnel.
+
+<p float="left">
+  <img src="Img/Max_L2VNI_Per_Tunnel.png" width="600"  hspace="50"/>
+</p>
+
+
+#### Test steps
+* Configure one L2VNI per leaf and check the functionality. Scale the config such that each spine has the same leaf connected and the tunnel to support max L2VNIs
+* Configure EBGP/IBGP as underlay protocol.
+* Configure EBGP as overlay protocol for remote mac learning.
+* Start all protocols.
+* Verify that switch is able to learn all remote VTEPs and measure the CPU utilization and memory usage.
+* Measure the time it takes to learn configured L2VNI's while doing scale testing.
+* Verify that different route types are learned and shown in database.
+* Send traffic from local host to remote hosts configured behind these VTEPs.
+* Match the tunnel counters with traffic statistics.
+* Enable egress tracking on vlan and see packets mapped between VNI and Vlan's.
+* Traffic should flow without any loss at line rate.
+* Do warm, fast, soft and cold reboots while traffic is running.
+
+#### Test results
+
+
+### Test case # 5 – Validate the functionality of L3VNI/VRF and the maximum L3VNI/VRF supported per switch - 512.
+#### Test objective
+Verify that swich supports upto 512 VRF instances.
+
+<p float="left">
+  <img src="Img/Max_VRF.png" width="600"  hspace="50"/>
+</p>
+
+
+#### Test steps
+* Validate the functionality of L3VNI/VRF by configuring single L3VNI on each leaf. Scale the config such that each leaf supports 512 VRFs per switch.
+* Configure EBGP/IBGP as underlay protocol.
+* Configure IBGP as overlay protocol for remote mac learning.
+* Start all protocols.
+* Verify that switch is able to learn all remote VTEPs and measure the CPU utilization and memory usage.
+* Measure the time it takes to learn configured L3VNI's while doing scale testing.
+* Verify that different route types are learned and shown in database.
+* Send traffic from local host to remote hosts configured behind these VTEPs.
+* Match the tunnel counters with traffic statistics.
+* Enable egress tracking on vlan and see packets mapped between VNI and Vlan's.
+* Traffic should flow without any loss at line rate.
+* Do warm, fast, soft and cold reboots while traffic is running.
+
+#### Test results
+
+
+### Test case # 6 – Host mac mobility from one VTEP to other.
+#### Test objective

```

Comment:
Does this test case covers MAC move from local to remote and remote to local?

Reply:
@skbhava yes.. plan is to cover vm/mac host mobility from local to remote and remote to local.